### PR TITLE
Pin golangci-lint to `v1.53.1` and upgrade `depguard` config to `v2`

### PR DESCRIPTION
**Pull request by tigrato**
Although our build box pin the `golangci-lint` script version during installation, the installed version itself wasn't pinned. The script pulls the latest release from GitHub and installs it. Yesterday, our build box was rebuilt, and ended up installing `golang-ci` `v1.53.0`. This version updated `depguard` to `v2` which introduced breaking changes for the `depguard` configuration format ([examples](https://github.com/golangci/golangci-lint/pull/3795/files)) that broke our lint process.

This PR converts `depguard` configuration to `v2` and pins the `golangci-lint` to `v1.53.1` to avoid future breaking changes.

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -298,7 +298,7 @@ RUN go install github.com/google/addlicense@v1.0.0
 RUN go install github.com/daixiang0/gci@v0.9.1
 
 # Install golangci-lint.
-RUN curl -fsSL "https://raw.githubusercontent.com/golangci/golangci-lint/v1.52.2/install.sh" | \
+RUN curl -fsSL "https://raw.githubusercontent.com/golangci/golangci-lint/v1.53.0/install.sh" | \
     sh -s -- -b "$(go env GOPATH)/bin"
```
Comment (marcoandredinis) [inline build.assets/Dockerfile:302]: As discussed in Slack, this will actually pull the latest version.
The recommended way in [golangci docs](https://golangci-lint.run/usage/install/#binaries) is to
```shell
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.52.2
```

However, using `master` does not sound like a great idea.
I think we should pin both the github tag and the binary version.
Something like this:
```suggestion
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/v1.53.0/install.sh | sh -s -- -b $(go env GOPATH)/bin v1.53.0
```

Comment (tigrato) [reply build.assets/Dockerfile]: Done in cbb6eb47f74b974d221b011c7a8cd5597feffc44

**Review by marcoandredinis — APPROVED** [review]

**Review by tcsc — APPROVED** [review]

File: build.assets/Dockerfile [hunk 2]
```diff
@@ -298,8 +298,10 @@ RUN go install github.com/google/addlicense@v1.0.0
 RUN go install github.com/daixiang0/gci@v0.9.1
 
 # Install golangci-lint.
-RUN curl -fsSL "https://raw.githubusercontent.com/golangci/golangci-lint/v1.52.2/install.sh" | \
-    sh -s -- -b "$(go env GOPATH)/bin"
+ARG GOLANGCI_VERSION # eg, "1.53.0"
+RUN TAG="v$GOLANGCI_VERSION" && \
+    curl -fsSL "https://raw.githubusercontent.com/golangci/golangci-lint/$TAG/install.sh" | \
+    sh -s -- -b $(go env GOPATH)/bin $TAG
```
Comment (codingllama) [inline build.assets/Dockerfile:304]: We don't need to pipe in the version from the Makefile, it's only used here.

```suggestion
RUN TAG='v.53.0' && \
    curl -fsSL "https://raw.githubusercontent.com/golangci/golangci-lint/$TAG/install.sh" | \
    sh -s -- -b $(go env GOPATH)/bin "$TAG"
```

File: build.assets/Makefile [hunk 3]
```diff
@@ -28,6 +28,9 @@ RUST_VERSION ?= 1.68.0
 LIBBPF_VERSION ?= 1.0.1
 LIBPCSCLITE_VERSION ?= 1.9.9-teleport
 
+# pin golangci-lint version to avoid breaking changes
+GOLANGCI_VERSION ?= 1.53.0
```
Comment (codingllama) [inline build.assets/Makefile:32]: Per comment above, I'd revert the changes here.

**Review by codingllama — COMMENTED** [review]
Thanks for fixing it. Please backport to active releases, we try to keep a consistent toolchain between the branches so there's less noise.

nit: s/deepguard/depguard/g

**Review by codingllama — APPROVED** [review]

**Comment by codingllama** [comment]
@tigrato asked me to apply the suggestions and move the PR forward, so that I shall.

**Comment by codingllama** [comment]
Buildbox and local runs seem both fine to me.
